### PR TITLE
Minor API tweaks. Add "Filename" synonym to connection string builder.

### PR DESCRIPTION
**Pull request by natemcmaster**
Add "Filename" alias to connection string builder
Rename CacheMode to SqliteConnectionCacheMode.

### CODE REVIEW

**Comment by divega** [comment]
Looks good to me but will leave it for @bricelam to signoff.

File: src/Microsoft.Data.Sqlite/SqliteConnectionStringBuilder.cs [hunk 1]
```diff
@@ -18,6 +18,7 @@ public class SqliteConnectionStringBuilder : DbConnectionStringBuilder
     {
         private const string DataSourceKeyword = "Data Source";
         private const string CacheKeyword = "Cache";
+        private const string FilenameKeyword = "Filename";
```
Comment (bricelam) [inline src/Microsoft.Data.Sqlite/SqliteConnectionStringBuilder.cs]: I think we also wanted *DataSource* to work.

**Comment by bricelam** [comment]
:shipit:

File: src/Microsoft.Data.Sqlite/SqliteConnectionStringBuilder.cs (hunk 1 above)
Comment (natemcmaster) [inline src/Microsoft.Data.Sqlite/SqliteConnectionStringBuilder.cs]: Will add.

**Pull request merged**